### PR TITLE
Upgrade kubernetes-client to 6.x

### PR DESCRIPTION
**Pull request by Vlatombe**
Downstream of
* https://github.com/jenkinsci/kubernetes-client-api-plugin/pull/163

- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by Vlatombe** [comment]
@maxlaverse Hi, are you around? I'll need soon to coordinate releases of various plugins including this one in order to upgrade `kubernetes-client` to a new major release. Would you be able to help?

**Comment by maxlaverse** [comment]
Hi @Vlatombe! Yes. I'm pretty sure I don't have a working local Java setup anymore but I'll look into it in the next days.

**Comment by Vlatombe** [comment]
@maxlaverse Otherwise if you are fine with it I can request co-maintainership on this plugin and then you don't have to worry  about it.

**Comment by maxlaverse** [comment]
Fine with me, but I'll have to get this working any way if I want to be able to merge&release https://github.com/jenkinsci/kubernetes-cli-plugin/pull/101

**Comment by Vlatombe** [comment]
Would next week (monday or tuesday be ok for you for coordinated releases?) I'll be making some announcements on the dev list to give a heads up.

**Comment by Vlatombe** [comment]
@maxlaverse Let me know when you're available for a merge/release of this PR. I'll hold on https://github.com/jenkinsci/kubernetes-client-api-plugin/pull/163 to minimize the time where not all parts are in sync.

**Comment by maxlaverse** [comment]
I don't have a strong preference @Vlatombe . I could do it in half an hour, 1.5 hour or this afternoon.

**Comment by Vlatombe** [comment]
Ok, I'll cut release of `kubernetes-client-api` now then.

**Comment by maxlaverse** [comment]
Should I wait for your signal or release this PR already @Vlatombe  ? And what about https://github.com/jenkinsci/kubernetes-cli-plugin/pull/101?

**Comment by Vlatombe** [comment]
CD release of `kubernetes-client-api` should be ongoing now. As soon as it is done you should be able to process this one. You can then do https://github.com/jenkinsci/kubernetes-cli-plugin/pull/101 as well.

File: pom.xml [hunk 1]
```diff
@@ -38,18 +38,17 @@
...
+    <jenkins.version>2.332.4</jenkins.version>
+    <bom.artifactId>bom-2.332.x</bom.artifactId>
+    <bom.version>1763.v092b_8980a_f5e</bom.version>
 
     <!-- dependency versions -->
     <authentication-tokens.version>1.3</authentication-tokens.version>
     <docker-commons.version>1.15</docker-commons.version>
     <google-oauth-plugin.version>0.8</google-oauth-plugin.version>
-    <kubernetes-client-api.version>5.4.1</kubernetes-client-api.version>
+    <kubernetes-client-api.version>6.3.1-205.vea_a_fc21de909</kubernetes-client-api.version>
```
Comment (Vlatombe) [inline pom.xml:51]:
```suggestion
    <kubernetes-client-api.version>6.3.1-206.v76d3b_6b_14db_b</kubernetes-client-api.version>
```

**Pull request merged**